test/cpp/ext/otel/otel_test_library.cc

```diff
@@ -38,11 +39,60 @@
 namespace grpc {
 namespace testing {
 
+#define GRPC_ARG_CSM_SERVICE_NAME "grpc.testing.csm_service_name"
+#define GRPC_ARG_CSM_SERVICE_NAMESPACE_NAME \
+  "grpc.testing.csm_service_namespace_name"
+
+// A subchannel filter that adds the service labels for test to the
+// CallAttemptTracer in a call.
+class AddServiceLabelsFilter : public grpc_core::ChannelFilter {
+ public:
+  static const grpc_channel_filter kFilter;
+
+  static absl::StatusOr<AddServiceLabelsFilter> Create(
+      const grpc_core::ChannelArgs& args, ChannelFilter::Args /*filter_args*/) {
+    return AddServiceLabelsFilter(
+        args.GetString(GRPC_ARG_CSM_SERVICE_NAME).value(),
+        args.GetString(GRPC_ARG_CSM_SERVICE_NAMESPACE_NAME).value());
+  }
+
+  grpc_core::ArenaPromise<grpc_core::ServerMetadataHandle> MakeCallPromise(
+      grpc_core::CallArgs call_args,
+      grpc_core::NextPromiseFactory next_promise_factory) override {
+    using CallAttemptTracer = grpc_core::ClientCallTracer::CallAttemptTracer;
+    auto* call_context = grpc_core::GetContext<grpc_call_context_element>();
+    auto* call_tracer = static_cast<CallAttemptTracer*>(
+        call_context[GRPC_CONTEXT_CALL_TRACER].value);
+    EXPECT_NE(call_tracer, nullptr);
+    auto service_labels =
+        std::make_shared<std::map<std::string, std::string>>();
```

Comment:
I think this can be written something like this:

```
auto service_labels =
    std::make_shared<std::map<std::string, std::string>>({
        {"service_name", service_name_},
        {"service_namespace", service_namespace_},
    });
```

Reply:
No longer needed.